### PR TITLE
Fixing container scoping and reusing service provider in WebJobs context

### LINKED ISSUE #9852: Fix DI scoping issue for HTTP requests in OOP branch

### PR DESCRIPTION
**Pull request by fabiocav**
### Issue describing the changes in this PR

resolves #9852 

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script/Host/ScriptInvocationArguments.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Script.Host
+{
+    public class ScriptInvocationArguments(IServiceProvider serviceProvider) : Dictionary<string, object>
```
Comment (mathewc) [inline src/WebJobs.Script/Host/ScriptInvocationArguments.cs]: Can this also be internal?

File: src/WebJobs.Script/DependencyInjection/InstanceServices/ScriptInstanceServicesProvider.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.WebJobs.Script.DependencyInjection
+{
+    internal class ScriptInstanceServicesProvider : IInstanceServicesProvider, IDisposable
```
Comment (mathewc) [inline src/WebJobs.Script/DependencyInjection/InstanceServices/ScriptInstanceServicesProvider.cs:10]: How hard would it be to add a regression test for this issue?

**Review by mathewc — APPROVED** [review]

File: src/WebJobs.Script/DependencyInjection/InstanceServices/ScriptInstanceServicesProvider.cs (hunk 2 above)
Comment (fabiocav) [reply src/WebJobs.Script/DependencyInjection/InstanceServices/ScriptInstanceServicesProvider.cs:10]: Planning on adding that prior to going into ‘dev’ as discussed offline. Main goal at the moment is getting the branch to a clean test/functional state and unblocking @kshyju ’s validations

File: src/WebJobs.Script/Host/ScriptInvocationArguments.cs (hunk 1 above)
Comment (fabiocav) [reply src/WebJobs.Script/Host/ScriptInvocationArguments.cs]: Yep. We can make that internal.

Comment (kshyju) [reply src/WebJobs.Script/Host/ScriptInvocationArguments.cs]: Updated the visibility of `ScriptInvocationArguments` to internal.

**Review by kshyju — APPROVED** [review]

**Pull request merged**